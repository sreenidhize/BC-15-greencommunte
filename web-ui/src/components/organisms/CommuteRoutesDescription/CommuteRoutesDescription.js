import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from "@material-ui/core";
import FiberManualRecordIcon from '@material-ui/icons/FiberManualRecord';
import RoomOutlinedIcon from '@material-ui/icons/RoomOutlined';
import CloseOutlinedIcon from '@material-ui/icons/CloseOutlined';

const useStyles = makeStyles({
    container: {
        maxWidth: "330px",
        height: "50px",
        margin: "5px 0px",
        position: "relative",
        borderRadius: "10px",
        background: "#e3f3f6",
    },
    location: {
        display: "flex",
        transform: "translateY(-50%)",
        position: "absolute",
        top: "50%",
    },
    font: {
        color: "#19293b",
        fontSize: "14px"
    },
    icon: {
        marginTop: "-3px",
        marginLeft: "8px",
        color: "#324552",
    },
    typoMargin: {
        marginLeft: "10px",
    },
    closeIcon: {
        position: "relative",
        right:"-20px",
    }

});

const CommuteRoutesDescription = ({ from, to }) => {
    const classes = useStyles();

    return (
        <div data-testid="commuteRoutesDescription-placeholder">
            <img src="https://i.stack.imgur.com/8kOXQ.jpg" alt="Commute Map" width="330px" height="135px"></img>
            <div className={classes.container} >
                <div className={classes.location}>
                    <FiberManualRecordIcon className={classes.icon} />
                    <Typography variant="h4" className={`${classes.font} ${classes.typoMargin}`}> {from||"Hyderabad, Telangana, India"} </Typography>
                    <CloseOutlinedIcon className={`${classes.icon} ${classes.closeIcon}`}/>
                </div>
            </div>
            <div className={classes.container}>
                <div className={classes.location}>
                    <RoomOutlinedIcon className={classes.icon} />
                    <Typography variant="h4" className={`${classes.font} ${classes.typoMargin}`}> {to||"Hyderabad, Telangana, India"} </Typography>
                </div>
            </div>
        </div >

    );
};

CommuteRoutesDescription.propTypes = {
    from: PropTypes.string,
    to: PropTypes.string,
};

export default CommuteRoutesDescription;