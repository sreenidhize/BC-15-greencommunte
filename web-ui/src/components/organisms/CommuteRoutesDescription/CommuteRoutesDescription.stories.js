import React from 'react';
import CommuteRoutesDescription from './CommuteRoutesDescription';
import theme from '../../../utils/theme/theme';
import { ThemeProvider } from '@material-ui/core';

export default {
    title: 'Organisms/RoutesDescription',
    component: CommuteRoutesDescription
};

const Template = (args) => <ThemeProvider theme={theme}><CommuteRoutesDescription {...args} /></ThemeProvider>;

export const Description = Template.bind({});
Description.args = {
    from: "East Marredpally, E Marredpally",
    to: "Hitech City, Telangana, Secunderabad",
};
