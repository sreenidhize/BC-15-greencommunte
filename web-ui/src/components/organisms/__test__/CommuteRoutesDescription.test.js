import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CommuteRoutesDescription from '../CommuteRoutesDescription/CommuteRoutesDescription';


describe("CommuteRoutesDescription placeholder test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <CommuteRoutesDescription />
    );
    expect(container).toMatchSnapshot();
  });

  test("test CommuteRoutesDescription render", () => {
    const { getByTestId } = render( <CommuteRoutesDescription />);
    expect(getByTestId("commuteRoutesDescription-placeholder")).toBeInTheDocument();
  });
});