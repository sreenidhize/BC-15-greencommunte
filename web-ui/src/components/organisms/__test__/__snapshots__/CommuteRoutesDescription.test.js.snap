// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommuteRoutesDescription placeholder test Matches the snapshot 1`] = `
<div>
  <div
    data-testid="commuteRoutesDescription-placeholder"
  >
    <img
      alt="Commute Map"
      height="135px"
      src="https://i.stack.imgur.com/8kOXQ.jpg"
      width="330px"
    />
    <div
      class="makeStyles-container-1"
    >
      <div
        class="makeStyles-location-2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="8"
          />
        </svg>
        <h4
          class="MuiTypography-root makeStyles-font-3 makeStyles-typoMargin-5 MuiTypography-h4"
        >
           
          Hyderabad, Telangana, India
           
        </h4>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-4 makeStyles-closeIcon-6"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
          />
        </svg>
      </div>
    </div>
    <div
      class="makeStyles-container-1"
    >
      <div
        class="makeStyles-location-2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9z"
          />
          <circle
            cx="12"
            cy="9"
            r="2.5"
          />
        </svg>
        <h4
          class="MuiTypography-root makeStyles-font-3 makeStyles-typoMargin-5 MuiTypography-h4"
        >
           
          Hyderabad, Telangana, India
           
        </h4>
      </div>
    </div>
  </div>
</div>
`;
